frappe_etax_service/models/account_move.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2023 Kitti U.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from odoo import models
+
+
+class AccountMove(models.Model):
+    _name = "account.move"
+    _inherit = ["account.move", "etax.th"]
+
+    def _get_branch_id(self):
+        """
+        By default, core odoo do not provide branch_id field in account.move and account.payment.
+        This method will check if branch_id is exist in model and return branch_id
+        """
+        if "branch_id" in self.env["account.move"]._fields:
+            return self.branch_id
+
+    def _prepare_inet_data(self, form_type="", form_name=""):
+        res = super()._prepare_inet_data()
+        branch_id = self._get_branch_id()
+        if branch_id:
+            res.update({"c02_seller_branch_id": branch_id.name})
+        if self.create_purpose_code:
```

Comment:
Why not just put this part in inet_data_template.py? 
เราควรจะไประวังให้มีการ assign self.create_purpose_code, self.create_purpose ถูกต้องมากกว่า
Note: inet_data_template.py จะได้เป็นที่เดียวที่เราดูและเรื่องความถูกต้องของข้อมูล (more simple)

Reply:
Oh I got it.
จริง ๆ เราเพิ่ม ข้อมูลไปใน inet_data_template ได้เลย ถ้า field หรือ method ไม่มีค่า ก็จะเป็นค่าว่าง ตามปกติ

I will fixed it kub.